kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -274,33 +313,103 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
       return null;
     }
 
-    int workerId = nextWorker.getAndIncrement() % workers.size();
-    KafkaConsumerWorker worker = workers.get(workerId);
-    return worker.readRecords(
-        state, timeout, maxBytes,
-        new ConsumerWorkerReadCallback<ClientKeyT, ClientValueT>() {
-          @Override
-          public void onCompletion(
-              List<? extends ConsumerRecord<ClientKeyT, ClientValueT>> records, Exception e
-          ) {
-            updateExpiration(state);
-            if (e != null) {
-              // Ensure caught exceptions are converted to RestExceptions so the user gets a
-              // nice error message. Currently we don't define any more specific errors because
-              // the old consumer interface doesn't classify the errors well like the new
-              // consumer does. When the new consumer is available we may be able to update this
-              // to provide better feedback to the user.
-              Exception responseException = e;
-              if (!(e instanceof RestException)) {
-                responseException = Errors.kafkaErrorException(e);
+    final KafkaConsumerReadTask task = new KafkaConsumerReadTask<KafkaKeyT, KafkaValueT,
```

Comment:
Should the `ReadCallback` parameter to this function be typed as `ReadCallback<ClientKeyT, ClientValueT>`?

Reply:
Seeing as both callbacks in `ConsumerManager` and `KafkaConsumerManager` expose the same interface, I think it makes sense to have both use one. So I'll rename the `WorkerReadCallback` and delete the local `ReadCallbacks`s